internal/app/app.go

```diff
@@ -67,7 +77,7 @@ func New(configFile string, logLevel int) (*Syncer, error) {
 		case common.VendorDell:
 			var dup vendors.Vendor
 
-			dup, err = dell.NewDUP(context.TODO(), firmwares, cfgSyncer, logger)
+			dup, err = dell.NewDUP(context.TODO(), firmwares, cfgSyncer, logger, v)
```

Comment:
instead of passing the viper pointer, it would be nicer (IMO) to parse this into one (or several?) structs that were specific to the vendors?